server/src/main/java/org/elasticsearch/search/aggregations/support/AggregationContext.java

```diff
@@ -221,6 +222,16 @@ public final AggregationUsageService getUsageService() {
      */
     public abstract SubSearchContext subSearchContext();
 
+    /**
+     *  Check if the given field is multivalued
+     *
+     * @param field Name of the field to check
+     * @return true if the field contians multiple indexed values
+     */
+    public boolean isMultiValued(String field) throws IOException {
+        return PointValues.size(searcher().getIndexReader(), field) > PointValues.getDocCount(searcher().getIndexReader(), field);
```

Comment:
This is only going to work on a points field - it should probably throw if the field doesn't have points. Because it's just not the right kind of field.

Reply:
I think it's worth keeping the size in a variable so you don't have to do it twice. It looks fairly fast but we may as well not do it twice.